src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: 'triangle' = 'triangle';
+
+  readonly color: 'red' | 'green' | 'blue';
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public readonly a: number,
+    public readonly b: number,
+    public readonly c: number,
+  ) {
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides must be greater than zero.');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('The given sides do not form a valid triangle.');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  readonly shape: 'circle' = 'circle';
 
-export class Circle implements Figure {}
+  readonly color: 'red' | 'green' | 'blue';
 
-export class Rectangle implements Figure {}
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public readonly radius: number,
+  ) {
+    this.color = color;
+
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than zero.');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100 - 0.2) / 100;
```

Comment:
The use of `Math.floor` with a subtraction of 0.2 for the circle's area calculation seems arbitrary. Consider using `Math.round` to round the area to the nearest hundredth for consistency across all shapes.